examples/gno.land/r/demo/boards2/public.gno

```diff
@@ -17,25 +18,49 @@ func GetBoardIDFromName(name string) (BoardID, bool) {
 func CreateBoard(name string) BoardID {
 	assertIsUserCall()
 
+	name = strings.TrimSpace(name)
+	if name == "" {
+		panic("board name is empty")
+	}
+
 	caller := std.GetOrigCaller()
 	assertIsNotAnonymousCaller(caller)
 
 	id := incGetBoardID()
-	board := newBoard(id, name, caller)
-	gBoardsByID.Set(id.Key(), board)
-	gBoardsByName.Set(name, board)
-	return board.id
+	args := Args{name, id}
+	// TODO: Do the callback really need the args or we could have the same result directly referencing?
+	gAuth.WithPermission(caller, PermissionBoardCreate, args, func(a Args) { // TODO: Caller could be an extra func arg
+		name := a[0].(string)
+		id := a[1].(BoardID)
+		board := newBoard(id, name, caller)
+		gBoardsByID.Set(id.Key(), board)
+		gBoardsByName.Set(name, board)
+	})
```

Comment:
Sending the arguments to `WithPermission` is necessary for the case where the `Permissioner` interface implementer, for example a DAO, creates a proposal under the hood. The arguments could be considered parameters for the proposal in that case.

Does using the args in the callback function make sense? Code would be cleaner if outer variables are used instead. Or is it better for the VM to use args from the function instead of outer references?

Caller is normally used within the callback, so we could send the caller as an argument for the callback if we don't want to use outer references.